src/waveform/renderers/waveformmark.cpp

```diff
@@ -5,21 +5,23 @@
 
 #include "waveformmark.h"
 
-WaveformMark::WaveformMark(int hotCue)
+WaveformMark::WaveformMark( const QString& group,
+                            const QDomNode& node,
+                            const SkinContext& context,
+                            const WaveformSignalColors& signalColors,
+                            int hotCue,
+                            QString item)
     : m_iHotCue(hotCue) {
-}
-
-void WaveformMark::reset(int hotCue) {
-    WaveformMark(hotCue).swap(*this);
-}
-
-void WaveformMark::setup(const QString& group, const QDomNode& node,
-                         const SkinContext& context,
-                         const WaveformSignalColors& signalColors) {
-    QString item = context.selectString(node, "Control");
+    if(item.isEmpty())
+        item = context.selectString(node, "Control");
```

Comment:
I don't understand this, how can I remove the if part from the constructor and put this to the part of the call in the waveformmarkset.cpp file, because when I call constructor in the waveformmarkset.cpp, I am giving the optional argument item there, which will be used, but in the case of no argument, the constuctor will generate the item argument by the context and use it to initialize the controlproxy pointer, so how can I remove this part from the constructor and put it on the calling part